worlds/shivers/__init__.py

```diff
@@ -0,0 +1,194 @@
+    """ 
+     Shivers is a horror themed point and click adventure. Explore the mysteries of Windlenot's Museum of the Strange and Unusual.
+    """
+
+    game: str = "Shivers"
+    topology_present = False
+    web = ShiversWeb()
+    option_definitions = Shivers_options
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = Constants.location_name_to_id
+    data_version = 0
+    
+    def create_item(self, name: str) -> Item:
+        data = item_table[name]
+        return ShiversItem(name, data.classification, data.code, self.player)
+
+    def create_event(self, region_name: str, event_name: str) -> None:
+        region = self.multiworld.get_region(region_name, self.player)
+        loc = ShiversLocation(self.player, event_name, None, region)
+        loc.place_locked_item(self.create_event_item(event_name))
+        region.locations.append(loc)
+
+    def create_regions(self) -> None:
+        # Create regions
+        for region_name, exits in Constants.region_info["regions"]:
+            r = Region(region_name, self.player, self.multiworld)
+            for exit_name in exits:
+                r.exits.append(Entrance(self.player, exit_name, r))
+            self.multiworld.regions.append(r)
+
+        # Bind mandatory connections
+        for entr_name, region_name in Constants.region_info["mandatory_connections"]:
+            e = self.multiworld.get_entrance(entr_name, self.player)
+            r = self.multiworld.get_region(region_name, self.player)
+            e.connect(r)
+        
+        # Locations
+        # Build exclusion list
+        include_information_plaques: bool = getattr(self.multiworld, "include_information_plaques")[self.player].value
+        elevators_stay_solved: bool = getattr(self.multiworld, "elevators_stay_solved")[self.player].value
+        early_lightning: bool = getattr(self.multiworld, "early_lightning")[self.player].value
+
+        self.removed_locations = set()
+        if not include_information_plaques:
+            self.removed_locations.update(Constants.exclusion_info["plaques"])
+        if not elevators_stay_solved:
+            self.removed_locations.update(Constants.exclusion_info["elevators"])
+        if not early_lightning:
+            self.removed_locations.update(Constants.exclusion_info["lightning"])
+
+        # Add locations
+        for region_name, locations in Constants.location_info["locations_by_region"].items():
+            region = self.multiworld.get_region(region_name, self.player)
+            for loc_name in locations:
+                if loc_name not in self.removed_locations:
+                    loc = ShiversLocation(self.player, loc_name, self.location_name_to_id.get(loc_name, None), region)
+                    region.locations.append(loc)
+
+    def create_items(self) -> None:
+        # Add pots
+        pots = [self.create_item(name) for name, data in item_table.items() if data.type == 'pot']
```

Comment:
it'd be much better to do all of these in a single loop since you don't seem to actually do any differentiation between all these lists. something like
```python
itempool = []
for name, data in item_table.items():
  if data.type in {"pot", "key", "ability", "filler2"}:
    itempool.append(self.create_item(name))```

Reply:
Updated to use suggested.